db/core/htrie.c

```diff
@@ -602,6 +718,11 @@ tdb_htrie_create_rec(TdbHdr *dbh, unsigned long off, unsigned long key,
 	if (data)
 		memcpy_fast(ptr, data, len);
 
+	bckt = (TdbBucket *)((unsigned long)r & TDB_HTRIE_DMASK);
+	bckt->flags |= (TDB_HTRIE_COMPLETE_BIT * complete);
```

Comment:
This is for variable length records only, right? This should be under `if (TDB_HTRIE_VARLENRECS(dbh))`.

Why this is for the whole bucket? If we have fully resolved key, we may have collisions in the bucket, so even for variable length records we can have many `TdbVRec`s in a bucket. I'd expect that the bit should be set for `TdbVRec`.